icu4j/main/tests/core/src/com/ibm/icu/dev/test/message2/Mf2FeaturesTest.java

```diff
@@ -0,0 +1,461 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.dev.test.message2;
+
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.ibm.icu.dev.test.TestFmwk;
+import com.ibm.icu.math.BigDecimal;
+import com.ibm.icu.util.Currency;
+import com.ibm.icu.util.CurrencyAmount;
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("javadoc")
+public class Mf2FeaturesTest extends TestFmwk {
+
+    // November 23, 2022 at 7:42:37.123 PM
+    static final Date TEST_DATE = new Date(1669261357123L);
+
+    @Before
+    public void prepareDefaultLocale() {
+        Locale.setDefault(Locale.US);
```

Comment:
Is this necessary?
If it is, then you should also later restore the default locale to what it used to be. Otherwise you change the environment for other ICU4J unit tests.

Reply:
Done.
Setting and restoring default locale in one method only.